jena-arq/src/main/java/org/apache/jena/riot/ResultSetMgr.java

```diff
@@ -36,123 +41,216 @@
+        Boolean b = readAny(urlOrFilename, lang).getBooleanResult();
+        return b;
+    }
+    
+    /** Read a boolean result from the URI
+     * 
+     * @param input
+     * @return boolean
+     */
+    public static boolean readBoolean(InputStream input) {
+        Boolean b = readAny(input).getBooleanResult();
+        return b;
+    }
+    
+    /** Read a boolean result from the URI;
+     * the input is expect to use syntax {@code lang}
+     * 
+     * @param input
+     * @param lang
+     * @return boolean
+     */
+    public static boolean readBoolean(InputStream input, Lang lang) {
+        Boolean b = readAny(input, lang).getBooleanResult();
+        return b;
     }
 
+    private static SPARQLResult readAny(String url) {
+        return ResultsReader.create().build().readAny(url);
+    }
+
+    private static SPARQLResult readAny(String url, Lang lang) {
+        checkLang(lang);
+        return ResultsReader.create()
+            .lang(lang)
+            .build()
+            .readAny(url);
+    }
+    
+    private static SPARQLResult readAny(InputStream input) {
+        return ResultsReader.create().build().readAny(input);
+    }
+
+    private static SPARQLResult readAny(InputStream input, Lang lang) {
+        checkLang(lang);
+        return ResultsReader.create()
+            .lang(lang)
+            .build()
+            .readAny(input);
+    }
+    // -------------------------------
+    
     /** Read ResultSet.
      * @param uri       URI to read from (includes file: and a plain file name).
      * @param hintLang  Hint for the syntax
      * @param context   Content object to control reading process.
      */
-    public static ResultSet parse(String uri, Lang hintLang, Context context)
-    {
-        // Conneg
-        if ( uri == null )
-            throw new IllegalArgumentException("URI to read from is null") ;
-        if ( hintLang == null )
-            hintLang = RDFLanguages.filenameToLang(uri) ;
-        TypedInputStream in = RDFDataMgr.open(uri, context) ;
-        if ( in == null )
-            throw new RiotException("Not found: "+uri) ;
-        return process(in, uri, hintLang, context) ;
-    }
-        
-    private static ResultSet process(TypedInputStream in, String srcURI, Lang hintLang, Context context) {
-        ContentType ct = WebContent.determineCT(in.getContentType(), hintLang, srcURI) ;
-        if ( ct == null )
-            throw new RiotException("Failed to determine the content type: (URI="+srcURI+" : stream="+in.getContentType()+" : hint="+hintLang+")") ;
-        ResultSetReader reader = getReader(ct) ;
-        if ( reader == null )
-            throw new RiotException("No parser registered for content type: "+ct.getContentType()) ;
-        return reader.read(in, context) ;
+    public static ResultSet parse(String uri, Lang hintLang, Context context) {
+        ResultSet rs = ResultsReader.create().lang(hintLang).context(context).read(uri);
```

Comment:
All read public methods of `ResultSetMgr` that take a `Lang` use `readAny` so `lang` gets checked.

Adding a check to the write methods makes sense (now done in local working copy) because it's necessary (no default included, no legacy requirement).

`ResultSetMgr`isn't the main API (`ResultSetFormatter` is, albeit older and clunkier).

For `parse`, the method is only there because there was one before this upgrade. There isn't a "no lang" version. lang==null makes sense because the `uri` is used (conneg or file extension).

Some of the `if ( rs == null )` checks are actually unnecessary because `SPARQLResult` does not return null, it throws an exception, but have been left in place in case that changes.

Reply:
:+1: makes sense. Happy to take another look once you say it's complete and ready to be merged. Thanks for the quick reply.

Reply:
I'll add checks to `ResultReader` and `ResultWriter` and use try-resource,